logstash-core/lib/logstash/environment.rb

```diff
@@ -1,7 +1,39 @@
 # encoding: utf-8
 require "logstash/errors"
+require "logstash/config/cpu_core_strategy"
+require "logstash/settings"
 
 module LogStash
+
+  [
+            Setting::String.new("node.name", Socket.gethostname),
+            Setting::String.new("config.path", nil, false),
+            Setting::String.new("config.string", nil, false),
+           Setting::Boolean.new("config.test", false),
+           Setting::Boolean.new("config.reload.auto", false),
+           Setting::Numeric.new("config.reload.interval", 3),
+           Setting::Boolean.new("metric.collect", true) {|v| v == true }, # metric collection cannot be disabled
+            Setting::String.new("settings.dir", ::File.join(Environment::LOGSTASH_HOME, "config")),
```

Comment:
> As a reminder, in logstash we use the notion of configuration for the pipeline description, so using LS_CONF_DIR can be misleading. Currently it's LS_SETTINGS_DIR

Correct. I vouch for `LS_SETTINGS_PATH` which is more precise for a settings file. I'd rather make it explicit there is just one settings.yml file we load, and hence the explicit path, rather than a dir.